in_toto/models/statement.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+# Copyright New York University and the in-toto contributors
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+<Program Name>
+  statement.py
+
+<Author>
+  Chasen Bettinger <bettingerchasen@gmail.com>
+
+<Started>
+  Mar 25, 2023
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Provides a class that represents a Statement, the middle layer of an 
+  attestation, binding it to a particular subject and unambiguously 
+  identifying the types of the Predicate as defined in the in-toto 
+  attestation spec.
+"""
+
+import attr
+
+import in_toto_attestation.v1.statement_pb2 as statementpb
+import in_toto_attestation.v1.resource_descriptor_pb2 as rdpb
+from google.protobuf.json_format import MessageToDict
+from in_toto.models.common import Signable
+
+STATEMENT_TYPE = "https://in-toto.io/Statement/v1"
+
+
+# NOTE: 08/24/23 - This function is an intentional
```

Comment:
@adityasaky @marcelamelara Review please

Reply:
I fear there is not much else you can do here, if you want Statement to be a Signable.

FYI: I had a very similar issue in in-toto-golang a while ago and tried the same fix you proposed in https://github.com/secure-systems-lab/securesystemslib/issues/625 but did not use it for the reasons pointed out by @jku. The fix I did end up using there ([handling special-case `json.Number`](https://github.com/in-toto/in-toto-golang/commit/814100d385dabf507399ff183eba16e3071db142)) won't work Python. But maybe that's okay, because the fix is actually quite a hack.

Because of all these problems with canonical json, I think the best way going forward would be to just not support a canonical json representation for Statements, and use it with DSSE Envelopes only.

Reply:
Seems like others have said this before: https://github.com/in-toto/in-toto/pull/570#discussion_r1213220358

Reply:
Yea, understood. I'm curious to learn the ramifications of adding a new method to the Envelope class within this library that doesn't require a `Signable` instance to instantiate.